commitlint.config.js

```diff
@@ -0,0 +1,71 @@
+module.exports = {
+    rules: {
+        // rules are defined as [ severity (2=error), "always"|"never", argument? ]. See
+        // https://commitlint.js.org/#/reference-rules?id=rules
+
+        // log-commit-info is used for debugging. It outputs the dict passed to a plugin.
+        // 'log-commit-info': [2, 'always'],
+        // enforce that commits with a breaking change indicator ("!") must also contain an explanation of the breaking change.
+        'breaking-change-indicator': [2, 'always'],
+        // enforce that only the types below are allowed
+        'type-enum': [2, 'always',
+            [
+                'feat',
+                'fix',
+                'dep',
+                'ci',
+                'refactor',
+                'doc',
+                'test',
+            ]
+        ],
+        // enforce that a type is mandatory
+        'type-empty': [2, 'never'],
+        // enforce the maximum subject length (without type and scope)
+        'subject-max-length': [2, 'always', 75],
+        // enforce the maximum scope length
+        'scope-max-length': [2, 'always', 25],
+    },
+    parserPreset: {
+        parserOpts: {
+            // add matching groups for `!` before and after scope. Before scope is invalid, after scope is correct
+            // and enforces the "breaking-change" header (see below).
+            headerPattern: /^(\w*)(?:(!?)\(([\w\$\.\-\* ]*)\))?(!?)\: (.*)$/,
+            headerCorrespondence: ["type", "wrong_breaking", "scope", "breaking", "subject"],
```

Comment:
```suggestion
            headerPattern: /^(\w*)(?:(!?)\(([\w\$\.\-\* ]*)\))?(!?)\:( ?)(.*)$/,
            headerCorrespondence: ["type", "wrong_breaking", "scope", "breaking", "subject_prefix", "subject"],
```

Reply:
This is going a bit overboard for me. The rule for breaking change indicators should not be reused for another check. If we can live with a slightly confusing error message when the user is formatting the commit wrong, then we should leave it as is.

Otherwise I’d rather start a new rule checker that makes sure that all necessary segments are there. I would not want to do that though.